types/codec_test.go

```diff
@@ -0,0 +1,39 @@
+package types
+
+import (
+	"encoding/json"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestJSONStripDefaults(t *testing.T) {
+	var x interface{}
+	err := json.Unmarshal([]byte(`{
+		"a":0,
+		"b":false,
+		"c":"",
+		"d":[],
+		"e":{},
+		"f":null,
+		"g":1,
+		"h":true,
+		"i":"abc",
+		"j":[0,false,"",[],{},1],
```

Comment:
add a case where we have an array of objects with empty/default value and non-empty/default values.